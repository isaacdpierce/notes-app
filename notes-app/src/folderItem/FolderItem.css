.folders__link {
  color: var(--main-p-color);
  text-decoration: none;
  text-transform: uppercase;
  letter-spacing: 2px;
  padding: 20px;
  transition: all 0.2s ease-in-out;
}
.folders__link:hover {
  color: white;
  background-color: var(--item-bg-hover-color);
}

.folders__item {
  color: var(--main-header-color);
  background-color: var(--item-bg-color);
  margin-bottom: 1rem;
  background-color: var(--item-bg-color);
  border-top: 1px solid var(--border-top-color);
  border-radius: 2px;
  box-shadow: 0 3px 1px 1px var(--box-shadow-color);
}

.folder-active {
  background-color: rgb(29, 29, 29);
}

.folder-active:hover {
  background-color: rgb(29, 29, 29);
}
