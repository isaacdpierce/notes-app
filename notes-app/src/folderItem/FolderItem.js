import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

import './FolderItem.css';

export default class FolderItem extends Component {
  render() {
    const { folder } = this.props;

    return (
      <li className='folders__item'>
        <NavLink
          to={`/folder/${folder.id}`}
          activeClassName='folder-active'
          className='folders__link'
        >
          {folder.name}
        </NavLink>
      </li>
    );
  }
}

FolderItem.propTypes = {
  folder: PropTypes.object,
};
