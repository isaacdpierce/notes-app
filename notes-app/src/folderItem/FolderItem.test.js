import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FolderItem from './FolderItem';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

it('renders a list item with navLink', () => {
  const folder = {
    id: '1234',
    name: 'Jimney Cricket',
  };
  const wrapper = shallow(<FolderItem folder={folder} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
