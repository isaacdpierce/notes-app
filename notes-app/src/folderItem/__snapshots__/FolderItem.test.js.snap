// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a list item with navLink 1`] = `
<li
  className="folders__item"
>
  <NavLink
    activeClassName="folder-active"
    className="folders__link"
    to="/folder/1234"
  >
    Jimney Cricket
  </NavLink>
</li>
`;
